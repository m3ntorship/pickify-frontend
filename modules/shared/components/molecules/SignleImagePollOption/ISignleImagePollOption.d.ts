import type { IPostFeed } from '@modules/shared/types/postFeed/IPostFeed';

declare namespace ISignleImagePollOption {
  export interface IProps {
    groupName: string;
    media: { url: string }[];
    options: IPostFeed.IOptions[];
    onOptionClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  }
}
export { ISignleImagePollOption };
