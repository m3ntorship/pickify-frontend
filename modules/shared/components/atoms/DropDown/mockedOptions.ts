export const options = [
  { id: 'delete', body: 'Delete Post' },
  { id: 'report', body: 'Report Post' },
  { id: 'save', body: 'Save Post' },
];

export const creationDropDown = [
  { id: 'delete', body: 'Delete Image' },
  { id: 'order', body: 'Change Order' },
];
