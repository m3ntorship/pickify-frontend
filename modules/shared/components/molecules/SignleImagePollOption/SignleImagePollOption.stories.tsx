import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import SignleImagePollOption from './SignleImagePollOption';
import type { ISignleImagePollOption } from './ISignleImagePollOption';

export default {
  component: SignleImagePollOption,
  title: 'Molecules/SignleImagePollOption',
};
const Template: Story<ISignleImagePollOption.IProps> = (args): ReactElement => (
  <SignleImagePollOption {...args} />
);
export const Default = Template.bind({});
Default.args = {
  groupName: 'caption',
  media: [{ url: 'eb519bfc-6f4c-4dfb-addc-75310e3945aa' }],
  options: [
    { body: 'yes', id: 'option_1', media: [] },
    { body: 'no', id: 'option_2', media: [] },
  ],
};
