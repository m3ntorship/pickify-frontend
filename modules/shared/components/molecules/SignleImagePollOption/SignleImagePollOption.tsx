import React from 'react';
import type { FC, ReactElement } from 'react';
import Image from 'next/image';
import { getVotesResults } from '../../../logic/votesLogic/votesLogic';
import type { ISignleImagePollOption } from './ISignleImagePollOption';
import ImagePollCovered from '../../atoms/ImagePollCovered/ImagePollCovered';
import ImageCaption from '../ImageCaption/ImageCaption';
import ImagePollUncovered from '../ImagePollUncovered/ImagePollUncovered';

const SignleImagePollOption: FC<ISignleImagePollOption.IProps> = ({
  groupName,
  media,
  options,
  onOptionClick,
}): ReactElement => {
  const { optionsPercentage } = getVotesResults(options);

  return (
    <div className="relative w-auto" data-testid="image-poll-option">
      {media.length !== 0 ? (
        <>
          {media.map((image) => (
            <div key={image.url} className="relative">
              <div className="absolute w-full h-full rounded-md overflow-hidden">
                <Image
                  src={image.url}
                  layout="responsive"
                  className="filter blur-sm"
                  objectFit="cover"
                  width={600}
                  height={600}
                />
              </div>
              <Image
                src={image.url}
                layout="responsive"
                className="rounded-md"
                objectFit="contain"
                width={600}
                height={600}
              />
            </div>
          ))}
          <div className="absolute flex bottom-4 right-4">
            {options.map((option, index) => (
              <div className="mr-4 last:mr-0" key={option.id}>
                {option.vote_count !== undefined ? (
                  <ImagePollUncovered
                    id={option.id}
                    percentage={optionsPercentage[index]}
                    optionBody={option.body}
                    type="circular"
                  />
                ) : (
                  <ImagePollCovered
                    isOneImageVote
                    like={option.body === 'yes'}
                    dislike={option.body === 'no'}
                    onOptionClick={onOptionClick}
                    id={option.id}
                  />
                )}
              </div>
            ))}
          </div>
          {groupName && (
            <div className="absolute bottom-4 left-4">
              <ImageCaption imgCaption={groupName} />
            </div>
          )}
        </>
      ) : (
        <div className="flex items-center justify-center">
          <p>cannot display this image :(</p>
        </div>
      )}
    </div>
  );
};

export default SignleImagePollOption;
