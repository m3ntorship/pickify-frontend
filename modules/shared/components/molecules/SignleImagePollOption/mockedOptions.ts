export const covoredOptions = [
  { body: 'yes', id: 'option_1', media: [] },
  { body: 'no', id: 'option_2', media: [] },
];

export const unCovoredOptions = [
  { body: 'yes', id: 'option_1', media: [], vote_count: 3 },
  { body: 'no', id: 'option_2', media: [], vote_count: 1 },
];
