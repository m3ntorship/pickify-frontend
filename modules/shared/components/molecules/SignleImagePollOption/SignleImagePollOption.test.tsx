import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SignleImagePollOption from './SignleImagePollOption';
import { covoredOptions, unCovoredOptions } from './mockedOptions';

describe('SignleImagePollOption', () => {
  it('should render covored like and dislike icons when passing covored options', () => {
    const tree = renderer
      .create(
        <SignleImagePollOption
          options={covoredOptions}
          imageUrl="https://source.unsplash.com/random"
          groupName="caption"
          onOptionClick={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render uncovored like and dislike icons when passing uncovored options', () => {
    const tree = renderer
      .create(
        <SignleImagePollOption
          options={unCovoredOptions}
          imageUrl="https://source.unsplash.com/random"
          groupName="caption"
          onOptionClick={(): boolean => true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
