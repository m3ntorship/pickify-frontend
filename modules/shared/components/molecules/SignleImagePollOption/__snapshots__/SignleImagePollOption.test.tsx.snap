// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignleImagePollOption should render covored like and dislike icons when passing covored options 1`] = `
<div
  className="relative w-auto"
  data-testid="image-poll-option"
>
  <div
    className="image-view-wrapper"
  >
    <div
      className="bg-image-box"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="group"
          className="bg-image"
          decoding="async"
          id="https://placeimg.com/640/480/any"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "100%",
          }
        }
      />
      <img
        alt="group"
        className="image"
        decoding="async"
        id="https://placeimg.com/640/480/any"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "contain",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="absolute flex bottom-4 right-4"
  >
    <div
      className="mr-4 last:mr-0"
    >
      <div
        className="flex justify-end"
      >
        <div
          className="flex"
        >
          <button
            className="like group"
            id="option_1"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="fill-primary stroke-current stroke-1 text-white group-hover:fill-white"
              data-file-name="SvgFilledLike"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="mr-4 last:mr-0"
    >
      <div
        className="flex justify-end"
      >
        <div
          className="flex"
        >
          <button
            className="dislike group"
            id="option_2"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="fill-error stroke-current stroke-1 text-white group-hover:fill-white"
              data-file-name="SvgFilledDislike"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="absolute bottom-4 left-4"
  >
    <div
      className="captionContent"
    >
      <span
        className="hidden"
      />
      <div
        className="captionDivider hidden"
      />
      <span
        className="image-caption"
        dir="auto"
        title="caption"
      >
        caption
      </span>
    </div>
  </div>
</div>
`;

exports[`SignleImagePollOption should render uncovored like and dislike icons when passing uncovored options 1`] = `
<div
  className="relative w-auto"
  data-testid="image-poll-option"
>
  <div
    className="image-view-wrapper"
  >
    <div
      className="bg-image-box"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="group"
          className="bg-image"
          decoding="async"
          id="https://placeimg.com/640/480/any"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "100%",
          }
        }
      />
      <img
        alt="group"
        className="image"
        decoding="async"
        id="https://placeimg.com/640/480/any"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "contain",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "visibility": "hidden",
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="absolute flex bottom-4 right-4"
  >
    <div
      className="mr-4 last:mr-0"
    >
      <div
        className="circular-vote-meter-wrapper"
        id="option_1"
      >
        <div
          className="circular-content"
        >
          <p
            className="mb-1"
          >
            üëç
          </p>
          <p>
            75
            %
          </p>
        </div>
        <div
          className="absolute "
        >
          <svg
            height={100}
            width={100}
          >
            <circle
              className="circular primary-text"
              cx={50}
              cy={50}
              fill="transparent"
              r={36}
              strokeDasharray="226.1946710584651 226.1946710584651"
              strokeWidth={7}
              style={
                Object {
                  "strokeDashoffset": "56.54866776461628",
                }
              }
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="mr-4 last:mr-0"
    >
      <div
        className="circular-vote-meter-wrapper"
        id="option_2"
      >
        <div
          className="circular-content"
        >
          <p
            className="mb-1"
          >
            üëé
          </p>
          <p>
            25
            %
          </p>
        </div>
        <div
          className="absolute "
        >
          <svg
            height={100}
            width={100}
          >
            <circle
              className="circular error-text"
              cx={50}
              cy={50}
              fill="transparent"
              r={36}
              strokeDasharray="226.1946710584651 226.1946710584651"
              strokeWidth={7}
              style={
                Object {
                  "strokeDashoffset": "169.64600329384882",
                }
              }
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="absolute bottom-4 left-4"
  >
    <div
      className="captionContent"
    >
      <span
        className="hidden"
      />
      <div
        className="captionDivider hidden"
      />
      <span
        className="image-caption"
        dir="auto"
        title="caption"
      >
        caption
      </span>
    </div>
  </div>
</div>
`;
